bindings/rascal/utils/io.py

```diff
@@ -0,0 +1,376 @@
+
+
+def is_large_array(data):
+    """is data a numpy array larger than 50MB ?"""
+    if is_npy(data):
+        if data.nbytes > 50e6:
+            return True
+        else:
+            return False
+    else:
+        return False
+
+
+def is_npy_filename(fn):
+    """does fn string corresponds to a saved numpy array ?"""
+    if isinstance(fn, str):
+        filename, file_extension = os.path.splitext(fn)
+        if file_extension == '.npy':
+            return True
+        else:
+            return False
+    else:
+        return False
+
+
+def get_class(module_name, class_name):
+    """Use module_name and class_name to make an instantiable class."""
+    module = importlib.import_module(module_name)
+    class_ = getattr(module, class_name)
+    return class_
+
+
+def obj2dict_beta(cls, state):
+    """Take a python object cls with its state and return a dictionary that
+    can be used to create a copy of this object.
+
+    Parameters
+    ----------
+    cls : object
+
+    state : dictionary
+        Contains the state of cls, i.e. the parameters used to initialize cls
+        in a 'init_params' field and the rest of the data needed to recover
+        the current state in a 'data' field.
+
+    Returns
+    -------
+    dictionary
+        fully serialized version of cls to a dictionary
+    """
+    VERSION = BETA_VERSION
+    module_name = cls.__module__
+    class_name = cls.__name__
+    frozen = dict(version=VERSION, class_name=class_name,
+                  module_name=module_name,
+                  init_params=state['init_params'],
+                  data=state['data'])
+    return frozen
+
+
+def dict2obj_beta(data):
+    """Take data, a dictionary created by the obj2dict function, and creates
+    a python object as described.
+
+    Parameters
+    ----------
+    data : dictionary
+        [description]
+
+    Returns
+    -------
+    deserialized python object described by data
+    """
+    cls = get_class(data['module_name'], data['class_name'])
+    obj = cls(**data['init_params'])
+    obj._set_data(data['data'])
+    return obj
+
+
+def is_valid_object_dict_beta(data):
+    """check compatibility of data to be used in dict2obj_beta"""
+    valid_keys = ['version', 'class_name', 'module_name', 'init_params', 'data']
+    aa = []
+    if isinstance(data, dict):
+        for k in data:
+            if k in valid_keys:
+                aa.append(True)
+        if len(aa) == len(valid_keys):
+            return True
+        else:
+            return False
+    else:
+        return False
+
+
+obj2dict = {BETA_VERSION: obj2dict_beta}
+dict2obj = {BETA_VERSION: dict2obj_beta}
+is_valid_object_dict = {BETA_VERSION: is_valid_object_dict_beta}
+
+
+def get_current_io_version():
+    return CURRENT_VERSION
+
+
+def get_supported_io_versions():
+    return list(dict2obj.keys())
+
+
+class BaseIO(ABC):
+    """
+    Definition of the interface of a python class that can be serialized by the
```

Comment:
[PEP 257](https://www.python.org/dev/peps/pep-0257/#multi-line-docstrings) specifies that the "summary line" of a multi-line docstring must fit on one line. This summary could be shortened e.g. to "Interface of a Python class serializable by to_dict()".  If a further description is necessary, it should be added below the summary, separated by a blank line.